### PR TITLE
p2p/sync: Wrong genesis digest is used to register pubsub topic subscriptions (race condition)

### LINKED ISSUE #5491: Initial sync took too long to switch to request parent sync
It took 13 mins from `Synced up to slot N` to `Requesting parent block`. What's up with the long delay?

Screen Shot 2020-04-18 at 8 47 01 AM

Took a profile during the delay. It doesn't say much:
Screen Shot 2020-04-18 at 8 16 49 AM

### PR DESCRIPTION
**Pull request by prestonvanloon**
Resolves #5491 #5476

### CODE REVIEW

File: beacon-chain/blockchain/service.go [hunk 1]
```diff
@@ -58,6 +58,7 @@ type Service struct {
 	headLock               sync.RWMutex
 	stateNotifier          statefeed.Notifier
 	genesisRoot            [32]byte
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/service.go:60]: What is the differnce between this and the one below?

Comment (nisdas) [reply beacon-chain/blockchain/service.go:60]: thats the genesis state root, which is different from the genesis validator root.

**Comment by prestonvanloon** [comment]
Thanks, LGTM @nisdas

**Comment by prestonvanloon** [comment]
works locally for me

**Comment by terencechain** [comment]
> Thanks, LGTM @nisdas

ditto

**Comment by terencechain** [comment]
> works locally for me

ditto. Thanks Nishant!

**Pull request merged**